connectors/sources/zoom.py

```diff
@@ -0,0 +1,540 @@
+                "sensitive": True,
+                "type": "str",
+                "value": "",
+            },
+            "fetch_past_meeting_details": {
+                "display": "toggle",
+                "label": "Fetch past meeting details",
+                "order": 4,
+                "tooltip": "Enable this option to fetch past past meeting details. This setting can increase sync time.",
+                "type": "bool",
+                "value": False,
+            },
+        }
+
+    async def validate_config(self):
+        await super().validate_config()
+        await self.client.api_token.get()
+
+    async def ping(self):
+        try:
+            await self.client.api_token.get()
+            self._logger.debug("Successfully connected to Zoom.")
+        except Exception:
+            self._logger.debug("Error while connecting to Zoom.")
+            raise
+
+    async def close(self):
+        await self.client.close()
+
+    def _format_doc(self, doc, doc_time):
+        doc = self.serialize(doc=doc)
+        doc.update(
+            {
+                "_id": doc["id"],
+                "_timestamp": doc_time,
+            }
+        )
+        return doc
+
+    def _pre_checks_for_get_content(
+        self, attachment_extension, attachment_name, attachment_size
+    ):
+        if attachment_extension == "":
+            self._logger.warning(
+                f"Files without extension are not supported, skipping {attachment_name}."
+            )
+            return False
+
+        if attachment_extension.lower() not in TIKA_SUPPORTED_FILETYPES:
+            self._logger.warning(
+                f"Files with the extension {attachment_extension} are not supported, skipping {attachment_name}."
+            )
+            return False
+
+        if attachment_size > FILE_SIZE_LIMIT:
+            self._logger.warning(
+                f"File size {attachment_size} of file {attachment_name} is larger than {FILE_SIZE_LIMIT} bytes. Discarding file content"
+            )
+            return
+        return True
+
+    async def _get_document_with_content(self, doc):
+        document = {
+            "_id": doc["id"],
+            "_timestamp": doc["date_time"],
+        }
+
+        temp_filename = ""
+
+        async with NamedTemporaryFile(mode="wb", delete=False) as async_buffer:
+            response = await self.client.get_file_content(
+                download_url=doc["download_url"]
+            )
+            await async_buffer.write(response.encode("utf-8"))
+            temp_filename = str(async_buffer.name)
+
+        await asyncio.to_thread(
+            convert_to_b64,
+            source=temp_filename,
+        )
+        async with aiofiles.open(file=temp_filename, mode="r") as target_file:
+            # base64 on macOS will add a EOL, so we strip() here
+            document["_attachment"] = (await target_file.read()).strip()
+        try:
+            await remove(temp_filename)
```

Comment:
This needs to be in a `finally` block. See other examples in other sources, like: https://github.com/elastic/connectors-python/blob/main/connectors/sources/s3.py#L195-L231